app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb

```diff
@@ -47,36 +48,30 @@ def set_identity_path_value
         def identity_path_resource_exists?
           return @identity_path_resource_exists unless @identity_path_resource_exists.nil?
 
-          @identity_path_resource_exists ||= !identity_path_resource.nil?
-        end
-
-        def identity_path_resource
-          @identity_path_resource ||= @resource_class[identity_path_resource_id]
-        end
-
-        def identity_path_resource_id
-          @identity_path_resource_id ||= "#{@authentication_parameters.authn_jwt_variable_id_prefix}/#{IDENTITY_PATH_RESOURCE_NAME}"
+          @identity_path_resource_exists ||= @check_authenticator_secret_exists.call(
+            conjur_account: @authentication_parameters.account,
+            authenticator_name: @authentication_parameters.authenticator_name,
+            service_id: @authentication_parameters.service_id,
+            var_name: IDENTITY_PATH_RESOURCE_NAME
+          )
         end
 
         def set_identity_path_from_policy
           @identity_path = identity_path_required_secret_value
         end
 
         def identity_path_required_secret_value
-          @identity_path_required_secret_value ||= identity_path_required_secret[identity_path_resource_id]
-        end
-
-        def identity_path_required_secret
-          @identity_path_required_secret ||= @fetch_required_secrets.call(resource_ids: [identity_path_resource_id])
+          @identity_path_required_secret_value ||= @fetch_authenticator_secrets.call(
+            conjur_account: @authentication_parameters.account,
+            authenticator_name: @authentication_parameters.authenticator_name,
+            service_id: @authentication_parameters.service_id,
+            required_variable_names: [IDENTITY_PATH_RESOURCE_NAME]
+          )[IDENTITY_PATH_RESOURCE_NAME]
         end
 
         def set_identity_path_default_value
```

Comment:
Is it still useful?

Reply:
Yes for case there is no identity-path in policy